mesonbuild/coredata.py

```diff
@@ -1030,14 +1034,17 @@ def add_lang_args(self, lang: str, comp: T.Type['Compiler'],
         # `self.options.update()`` is perfectly safe.
         self.options.update(compilers.get_global_options(lang, comp, for_machine, env))
 
-    def process_new_compiler(self, lang: str, comp: 'Compiler', env: 'Environment') -> None:
+    def process_compiler_options(self, lang: str, comp: 'Compiler', env: 'Environment') -> None:
         from . import compilers
 
         self.add_compiler_options(comp.get_options(), lang, comp.for_machine, env)
 
         enabled_opts: T.List[OptionKey] = []
         for key in comp.base_options:
             if key in self.options:
+                if key in env.options:
+                    # Override from commandline, when reconfiguring
+                    self.options[key].set_value(env.options[key])
```

Comment:
Maybe I'm missing something here, but couldn't this just be `if key in self.options and key not in env.options:`?

Reply:
You are right, it can be simplified. I tried something else...